packages/template-retail-react-app/app/components/product-view/index.jsx

```diff
@@ -55,7 +55,9 @@ const ProductViewHeader = ({name, currency, priceData, category, product}) => {
             </Skeleton>
 
             <Skeleton isLoaded={priceData?.currentPrice}>
-                <DisplayPrice priceData={priceData} currency={currency} />
+                {priceData?.currentPrice && (
+                    <DisplayPrice priceData={priceData} currency={currency} />
+                )}
```

Comment:
Why did we have to add this conditional here? It's implying that `priceData` can be defined but the value `currentPrice` inside of it might not? What scenario would this occur? Could that be handled internally in the DisplayPrice component?

Reply:
Since `priceData` is returned from `getDataPrice` which returns an object includes prices data (see logic in ProductView below). When product is undefined, the getPriceData will return object with currentPrice as `undefined`. 
```
const priceData = {
  currentPrice: undefined,
  listPrice: undefined,
  ...
}
```

For DisplayPrice, we want to make sure the price is showing at least one price, so we enforced `currentPrice`  to be required prop. I don't think it makes sense to handle this internally because if none of the price is available, we should not need to render DisplayPrice yet.